operator/src/main/java/oracle/kubernetes/operator/helpers/JobStepContext.java

```diff
@@ -389,6 +394,22 @@ protected V1Container createPrimaryContainer(TuningParameters tuningParameters)
     return container;
   }
 
+  private String getVolumeName(String resourceName, String type) {
+    return getName(resourceName, type);
+  }
+
+  private String getName(String resourceName, String type) {
+    return resourceName.length() > (MAX_ALLOWED_VOLUME_NAME_LENGTH - VOLUME_NAME_SUFFIX.length())
+            ? getShortName(resourceName, type)
+            : resourceName + VOLUME_NAME_SUFFIX;
+  }
+
+  private String getShortName(String resourceName, String type) {
+    String volumeSuffix = VOLUME_NAME_SUFFIX + "-" + type + "-"
+            + Optional.ofNullable(ChecksumUtils.getMD5Hash(resourceName)).orElse("");
```

Comment:
When would the result of getMD5Hash be null?

Reply:
It can be null when `MessageDigest.getInstance("MD5")` method (in ChecksumUtils) throws `NoSuchAlgorithmException` or `String.getBytes()` method throws `UnsupportedEncodingException`.